arrow/arrow.py

```diff
@@ -151,7 +151,7 @@ def fromdatetime(cls, dt, tzinfo=None):
 
         '''
 
-        tzinfo = tzinfo or dt.tzinfo or dateutil_tz.tzutc()
+        tzinfo = tzinfo if tzinfo is not None else dt.tzinfo or dateutil_tz.tzutc()
```

Comment:
I'm not certain what the order of operations is between `if` and `or`, and I expect most people won't either.  I feel this should have brackets around the `or`: `tzinfo if tzinfo is not None else (dt.tzinfo or dateutil_tz.tzutc())`.

Reply:
Actually, that would just have the same problem - if you are going to support falsy time-zones, the `dt.tzinfo` also need to be expanded into a `None` check.

Reply:
That said, disregarding falsy-`tzinfo`, the operator precedence doesn't actually matter in this case, because it's equivalent to a chained `or`:

1. `(tzinfo if tzinfo is not None else dt.tzinfo) or `dateutil_tz.tzutc()` 

2. `tzinfo if tzinfo is not None else (dt.tzinfo or `dateutil_tz.tzutc())` 

In both cases it will just use whichever the first one in the list is that is not `None`.

Reply:
I'd be OK with that, as it ensures everything is explicit, but we know `dt.tzinfo` is either a `tzinfo` object or `None`, so it's not strictly necessary.

Unless we're concerned about someone creating a `tzinfo` subclass that can be falsy?  I took a look at the `datetime` docs and, while it states that `datetime`, `date`, and `time` are always truthy, it doesn't define that behaviour for `tzinfo`.  So, yeah, this might actually be required.

Reply:
In `(tzinfo if tzinfo is not None else dt.tzinfo) or dateutil_tz.tzutc()`, if `tzinfo` is `0`, then `dateutil_tz.tzutc()` will be chosen.

With `tzinfo if tzinfo is not None else (dt.tzinfo or dateutil_tz.tzutc())`, if `tzinfo` is `0` then `dt.tzinfo` gets chosen.

Reply:
I said "disregarding falsy `tzinfo`". In the fixed version, passing `tzinfo=0` will be an error anyway. I was just saying that if you are operating under the assumption that `tzinfo` is an actual `datetime.tzinfo` (as the code has been up until now), the two are equivalent.